Core/test/qa-functional/src/org/sleuthkit/autopsy/ingest/EmbeddedFileTest.java

```diff
@@ -0,0 +1,194 @@
+    private static final File CASE_DIR = new File(CASE_DIRECTORY_PATH.toString());
+    private final Path IMAGE_PATH = Paths.get(this.getDataDir().toString(),"embedded.vhd");
+    public static final String HASH_VALUE = "098f6bcd4621d373cade4e832627b4f6";
+    private static final int DEEP_FOLDER_COUNT = 25;
+    private Case openCase;
+  
+    public static Test suite() {
+        NbModuleSuite.Configuration conf = NbModuleSuite.createConfiguration(EmbeddedFileTest.class).
+                clusters(".*").
+                enableModules(".*");
+        return conf.suite();
+    }
+
+    public EmbeddedFileTest(String name) {
+        super(name);
+    }
+
+    @Override
+    public void setUp() {
+        // Delete the test directory, if it exists
+        if (CASE_DIRECTORY_PATH.toFile().exists()) {
+            try {
+                FileUtils.deleteDirectory(CASE_DIRECTORY_PATH.toFile());
+            } catch (IOException ex) {
+                Exceptions.printStackTrace(ex);
+                Assert.fail(ex);
+            }
+        }
+        assertFalse("Unable to delete existing test directory", CASE_DIRECTORY_PATH.toFile().exists());
+ 
+        // Create the test directory
+        CASE_DIRECTORY_PATH.toFile().mkdirs();
+        assertTrue("Unable to create test directory", CASE_DIRECTORY_PATH.toFile().exists());
+
+        try {
+            Case.createAsCurrentCase(Case.CaseType.SINGLE_USER_CASE, CASE_DIRECTORY_PATH.toString(), new CaseDetails("EmbeddedFileTest"));
+        } catch (CaseActionException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        }
+        assertTrue(CASE_DIR.exists());
+        ImageDSProcessor dataSourceProcessor = new ImageDSProcessor();
+        try {
+            DataSourceProcessorRunner.ProcessorCallback callBack = DataSourceProcessorRunner.runDataSourceProcessor(dataSourceProcessor, IMAGE_PATH);
+            List<Content> dataSourceContent = callBack.getDataSourceContent();
+            assertEquals(1, dataSourceContent.size());
+            List<String> errorMessages = callBack.getErrorMessages();
+            assertEquals(0, errorMessages.size());
+        } catch (AutoIngestDataSourceProcessor.AutoIngestDataSourceProcessorException | InterruptedException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+            
+        }
+        
+        try {
+            openCase = Case.getOpenCase();
+            runIngestJob(openCase.getDataSources());
+        } catch (NoCurrentCaseException | TskCoreException ex) {
+            Exceptions.printStackTrace(ex);
+            Assert.fail(ex);
+        } 
+    }
+
+    @Override
+    public void tearDown() {
+        try {
+            Case.closeCurrentCase();
+            //Seems like we need some time to close the case.
+            try {
+                Thread.sleep(2000);
+            } catch (Exception ex) {
+
+            }
```

Comment:
I don't think this should be necessary. What did you observe that led you to add this sleep?